src/hotspot/share/gc/shared/space.hpp

```diff
@@ -548,6 +550,9 @@ class ContiguousSpace: public CompactibleSpace {
   // Size computations: sizes in bytes.
   size_t capacity() const        { return byte_size(bottom(), end()); }
   size_t used() const            { return byte_size(bottom(), top()); }
+  size_t live() const            {
```

Comment:
I would prefer if the code for serial gc, contrary to others, tries to give some resemblance of tracking actual liveness. I.e. calculating this anew every call to `SerialHeap::live()`.
If calling an `update_live_estimate()` in parallel and G1 (and the other collectors) is fine at certain places, this should be as good for serial gc.
Doing so would reduce the footprint of this change quite a bit (for serial gc)

Reply:
Ok. I am caching the live estimate per memory space now. 
Not sure how much it will change the footprint of this change, though, but it is good for consistency anyway.